abad.go

```diff
@@ -4,30 +4,56 @@ import (
 	"fmt"
 
 	"github.com/NeowayLabs/abad/ast"
+	"github.com/NeowayLabs/abad/builtins"
+	"github.com/NeowayLabs/abad/internal/utf16"
 	"github.com/NeowayLabs/abad/parser"
-	"github.com/NeowayLabs/abad/types"
 	"github.com/NeowayLabs/abad/token"
+	"github.com/NeowayLabs/abad/types"
 )
 
 type (
-	// Obj types
-	Obj interface {
-		String() string
-	}
-
 	// Abad interpreter, a very bad one.
 	Abad struct {
 		filename string
+
+		global *types.DataObject
 	}
 )
 
-func NewAbad(filename string) *Abad {
-	return &Abad{
+var (
+	consoleAttr = utf16.S("console")
+)
+
+func NewAbad(filename string) (*Abad, error) {
+	ecma := &Abad{
```

Comment:
It seems that "a" or "abad" would be more consistent with the rest of the code than "ecma"